source/common/compressor/compressor_impl.h

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include "envoy/compressor/compressor.h"
+
+#include "zlib.h"
+
+namespace Envoy {
+namespace Compressor {
+
+/**
+ * Implementation of compressor's interface.
+ */
+class CompressorImpl : public Compressor {
+public:
+  CompressorImpl();
+  ~CompressorImpl();
+
+  /**
+   * Enum values used to set compression level during initialization
+   * best: gives best compression.
+   * speed: gives best.
+   * standard: gives normal compression. (default)
+   */
+  enum class CompressionLevel : int8_t {
+    best = 9,
```

Comment:
nit: in general in Envoy enum values are capitalized. e.g., 'Best', 'Speed', etc. Same below.